src/com/xilinx/rapidwright/design/tools/RelocationTools.java

```diff
@@ -0,0 +1,272 @@
+                if (!c.isLocked() && !cells.contains(c)) {
+                    System.out.println("ERROR: Failed to relocate SiteInst '" + si.getName()
+                            + "' as it contains Cells both inside and outside of '" + instanceName + "'");
+                    error = true;
+                }
+            }
+        }
+
+        return !error && relocate(design, siteInsts, tileColOffset, tileRowOffset);
+    }
+
+    /**
+     * Relocate all SiteInsts (and PIPs) within the Pblock in-place by tileColOffset/tileRowOffset tiles.
+     *
+     * @param design Parent design
+     * @param pblock PBlock
+     * @param tileColOffset Relocate this number of tile columns (X axis)
+     * @param tileRowOffset Relocate this number of tile rows (Y axis)
+     * @return True if successful, false otherwise.
+     */
+    public static boolean relocate(Design design,
+                                   PBlock pblock,
+                                   int tileColOffset,
+                                   int tileRowOffset) {
+        Collection<SiteInst> siteInsts = new ArrayList<>();
+        for (Site s : pblock.getAllSites(null)) {
+            SiteInst si = design.getSiteInstFromSite(s);
+            if (si != null) {
+                siteInsts.add(si);
+            }
+        }
+        return relocate(design, siteInsts, tileColOffset, tileRowOffset);
+    }
+
+    /**
+     * Relocate all given SiteInsts and PIPs in-place by
+     * tileColOffset/tileRowOffset tiles.
+     *
+     * Should any SiteInst or PIP not be relocatable to a tile at the specified
+     * offset (e.g. if a compatible tile does not exist) function will return
+     * false and design will be unmodified.
+     *
+     * Any net sourced from a SiteInst not in the given set will be fully
+     * unrouted; those destined for a SiteInst not in the given set will have
+     * their specific branch unrouted.
+     *
+     * @param design Parent design
+     * @param siteInsts List of SiteInsts to be relocated
+     * @param tileColOffset Relocate this number of tile columns (X axis)
+     * @param tileRowOffset Relocate this number of tile rows (Y axis)
+     * @return True if successful, false otherwise.
+     */
+    public static boolean relocate(Design design,
+                                   Collection<SiteInst> siteInsts,
+                                   int tileColOffset,
+                                   int tileRowOffset) {
+        if (siteInsts.isEmpty())
+            return true;
+
+        if (tileColOffset == 0 && tileRowOffset == 0)
+            return true;
+
+        Map<SiteInst, Site> oldSite = new HashMap<>();
+        for (SiteInst si : siteInsts) {
+            assert(si.isPlaced());
+            oldSite.put(si, si.getSite());
+            si.unPlace();
+        }
+
+        boolean revertPlacement = false;
+        for (Map.Entry<SiteInst, Site> e : oldSite.entrySet()) {
+            Site ss = e.getValue();
```

Comment:
The variable names `ss, st, dt, ds, ssi, dsi`, I think they mean `srcSite, srcTile, dstTile, dstSite, srcSiteInst, dstSiteInst`, is that correct?  If so, probably no change needed.